### ISSUE TITLE
"Account Not Set Up" error after adding MSAL option

### ISSUE THREAD

**Issue by ikdkuh** [msintuneappsdk/ms-intune-app-sdk-ios#356]
(Please forgive me for writing using a translation tool.)
We use the Intune SDK for our in-house apps.
We have been informed that the policy will not be available unless the application is registered with Azure AD, and we are working on countermeasures internally.
After registering the app in Azure AD, I set the following items in the info.plist of the app and built it.
[IntuneMAMSettings]
- ADALAuthority : https://login.microsoftonline.com/[tenantId]
- ADALClientId : [ClientId of the app registered in Azure AD]
- ADALRedirectUri : msauth.[bundle ID]://auth

But when I transfer files from OneDrive to the app I get an error.
![AccountNotSetUp](https://github.com/msintuneappsdk/ms-intune-app-sdk-ios/assets/77658581/ac8d6221-8a88-44c3-aa5c-8c6e785684af)
Transferring between apps works fine if you don't set IntuneMAMSettings.
I also tried the settings described here, but that didn't work either.
(https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-get-started#give-your-app-access-to-the-intune-app-protection-service-optional)
Do I need to configure anything else?
Our in-house app uses Intune SDK Version 17.3.0.

Thanks.

**Comment by vigavadi** [comment]
@ikdkuh Are policies targeted to your app and the account that you are trying to use? https://learn.microsoft.com/en-us/mem/intune/apps/app-protection-policies

**Comment by ikdkuh** [comment]
@vigavadi 
Yes, I have an Intune policy targeted to the Bundle ID of this app and the users who use the app.
Also, the policy is still working normally for versions that do not define IntuneMAMSettings.
Any ideas what could be causing this error to appear?

**Comment by vigavadi** [comment]
@ikdkuh If the settings aren't defined then it uses the default Intune settings. That's why it works. 
Couple of things to check:
1. Have you properly configured MSAL: https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-ios#configure-msal
2. You need to give access to the MAM resource for your AAD registration: https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-get-started#give-your-app-access-to-the-intune-app-protection-service-optional

**Comment by vigavadi** [comment]
Closing due to inactivity.
